### DISCUSSION TITLE
关于theme-default的样式问题

### DISCUSSION THREAD

**Discussion by customCJY** [vuepress/core#928] [Q&A] [answered]
我使用vuepress以及自带的theme-default构建了组件库文档，但在编写组件demo的过程中发现组件样式会受到影响，排查发现是由于packages/@vuepress/theme-default/src/client/styles/normalize.scss这个文件下添加了一些元素全局样式重置了我的组件，例如：
`th,
td {
  padding: 0.6em 1em;
  border: 1px solid var(--c-border-dark);
  transition: border-color var(--t-color);
}
h2 {
  font-size: 1.65rem;
  padding-bottom: 0.3rem;
  border-bottom: 1px solid var(--c-border);
  transition: border-color var(--t-color);
}
`
这些样式就完全影响了我组件的样式，这个有什么好的解决方案么？在组件内部覆盖也不是一个很好的选择

**Comment by Mister-Hope** [comment, marked as answer]
> 在组件内部覆盖也不是一个很好的选择

- 在组件内部覆盖也不是一个很好的选择 ❌
- 组件库的组件必须 normalize 基础标签，避免浏览器内置央视或全局样式对组件显示产生影响。✔️
- 在有 normalize 行为的框架内，组件应自己处理 normalize 的影响。✔️

**Reply by customCJY** [reply]
我认为像` th, td { padding: 0.6em 1em; border: 1px solid var(--c-border-dark); transition: border-color var(--t-color); } h2 { font-size: 1.65rem; padding-bottom: 0.3rem; border-bottom: 1px solid var(--c-border); transition: border-color var(--t-color); }`中td元素的padding和h2元素的border-bottom样式更应该通过类名去定义，而不应该是normalize样式

**Reply by Mister-Hope** [reply]
。。。
这几行是为了夜间模式去nomalize标签的，它没有任何问题。

**Reply by customCJY** [reply]
夜间模式去nomalize标签？不是很明白，他影响到了具体组件的样式，跟日间或则夜间模式并无关系，所有组件中的td元素都被加上了padding: 0.6em 1em;的边距
